src/controllers/api-key.js

```diff
@@ -10,82 +10,259 @@
  * governing permissions and limitations under the License.
  */
 
-import { createResponse } from '@adobe/spacecat-shared-http-utils';
+import { createResponse, hashWithSHA256, ok } from '@adobe/spacecat-shared-http-utils';
+import { hasText, isObject, isValidUrl } from '@adobe/spacecat-shared-utils';
+import crypto from 'crypto';
 import { ErrorWithStatusCode } from '../support/utils.js';
+import {
+  STATUS_INTERNAL_SERVER_ERROR,
+  STATUS_BAD_REQUEST,
+  STATUS_CREATED,
+  STATUS_FORBIDDEN,
+  STATUS_NO_CONTENT,
+  STATUS_UNAUTHORIZED, STATUS_NOT_FOUND,
+} from '../utils/constants.js';
+import { ApiKeyDto } from '../dto/api-key.js';
 
 /**
  * ApiKey Controller. Provides methods for managing API keys such as create, delete, and get.
- * @returns {Object} ApiKey Controller
+ * @param {object} context - The context of the universal serverless function.
+ * @param {object} context.env - Environment details.
+ * @param {object} context.env.API_KEY_CONFIGURATION - Configuration for the API key controller.
+ * @param {DataAccess} context.dataAccess - Data access.
+ * @param {Logger} context.log - Logger.
+ * @param {object} context.attributes - Attributes that include the user's profile details
+ * @param {ImsClient} context.imsClient - IMS Client.
+ * @returns {object} ApiKey Controller
  * @constructor
  */
 function ApiKeyController(context) {
-  // eslint-disable-next-line no-unused-vars
-  const { log } = context;
+  const {
+    dataAccess, log, env, attributes, imsClient,
+  } = context;
+
+  let apiKeyConfiguration = {};
+  try {
+    apiKeyConfiguration = JSON.parse(env.API_KEY_CONFIGURATION);
+  } catch (error) {
+    log.error(`Failed to parse API Key configuration: ${error.message}`);
+  }
+
+  const { maxDomainsPerApiKey = 1, maxApiKeys = 3 } = apiKeyConfiguration;
   const HEADER_ERROR = 'x-error';
 
   function createErrorResponse(error) {
-    return createResponse({}, error.status, {
+    return createResponse({}, error.status || STATUS_INTERNAL_SERVER_ERROR, {
       [HEADER_ERROR]: error.message,
     });
   }
 
-  // eslint-disable-next-line no-unused-vars
+  /**
+   * Validate the request data.
+   * @type {{features: string[], domains: string[], name: string}}
```

Comment:
Is this used? I don't have experience with `@type`.

Reply:
I noticed it being used in other places in the api-service.